src/librustc_middle/ty/sty.rs

```diff
@@ -2210,21 +2210,28 @@ impl<'tcx> Const<'tcx> {
     /// Literals and const generic parameters are eagerly converted to a constant, everything else
     /// becomes `Unevaluated`.
     pub fn from_anon_const(tcx: TyCtxt<'tcx>, def_id: LocalDefId) -> &'tcx Self {
-        debug!("Const::from_anon_const(id={:?})", def_id);
+        Self::const_arg_from_anon_const(tcx, ty::WithOptParam::dummy(def_id))
+    }
+
+    pub fn const_arg_from_anon_const(
```

Comment:
Should probably be named `from_const_arg`.

Reply:
Okay you use `opt_const_arg` elsewhere, I think that would make sense here.

Reply:
Renamed to `from_opt_const_arg_anon_const` for now